opennms-services/src/test/java/org/opennms/netmgt/poller/DeviceConfigPersistenceIT.java

```diff
@@ -0,0 +1,148 @@
+        "classpath:/META-INF/opennms/applicationContext-deviceConfigDao.xml",
+        "classpath:/META-INF/opennms/applicationContext-pollerdTest.xml",
+        "classpath:/META-INF/opennms/mockEventIpcManager.xml"})
+@JUnitConfigurationEnvironment
+@JUnitTemporaryDatabase(reuseDatabase = false)
+public class DeviceConfigPersistenceIT {
+
+    @Autowired
+    private QueryManager queryManager;
+
+    @Autowired
+    private IpInterfaceDao ipInterfaceDao;
+
+    @Autowired
+    private NodeDao nodeDao;
+
+    @Autowired
+    private DeviceConfigDao deviceConfigDao;
+
+    private OnmsIpInterface ipInterface;
+    private OnmsNode node;
+
+    @Before
+    public void init() {
+        populateIpInterfaceAndGet();
+    }
+
+    @Test
+    public void testDeviceConfigPersistence() {
+        int count = 10;
+        String config = "OpenNMS-Device-Config";
+        // Send charset information through attributes.
+        byte[] configInBytes = config.getBytes(StandardCharsets.UTF_16);
+        populateDeviceConfigs(count);
+        PollableService service = Mockito.mock(PollableService.class);
+        Mockito.when(service.getNodeId()).thenReturn(node.getId());
+        Mockito.when(service.getIpAddr()).thenReturn(InetAddressUtils.toIpAddrString(ipInterface.getIpAddress()));
+        Map<String, Object> attributes = new HashMap<>();
+        attributes.put("encoding", StandardCharsets.UTF_16.name());
+        queryManager.persistDeviceConfig(service, attributes, configInBytes);
+        Optional<DeviceConfig> deviceConfigOptional = deviceConfigDao.getLatestConfigForInterface(ipInterface);
+        Assert.assertTrue(deviceConfigOptional.isPresent());
+        byte[] retrievedConfigInBytes = deviceConfigOptional.get().getConfig();
+        // Compare binary values
+        Assert.assertArrayEquals(configInBytes, retrievedConfigInBytes);
+        // Check that version should be latest
+        Assert.assertThat(deviceConfigOptional.get().getVersion(), Matchers.is(count + 1));
+        // Make use of encoding
+        String retrievedConfig = new String(retrievedConfigInBytes, Charset.forName(deviceConfigOptional.get().getEncoding()));
+        Assert.assertEquals(config, retrievedConfig);
+
+    }
+
+    private void populateDeviceConfigs(int count) {
+
+        for (int i = 1; i <= count; i++) {
+            DeviceConfig deviceConfig = new DeviceConfig();
+            deviceConfig.setIpInterface(ipInterface);
+            deviceConfig.setConfig(UUID.randomUUID().toString().getBytes(StandardCharsets.US_ASCII));
+            deviceConfig.setEncoding("ASCII");
+            // Persist old configs with time 1 day before.
+            deviceConfig.setCreatedTime(Date.from(Instant.now().minus(1, ChronoUnit.DAYS).plusSeconds(i * 60)));
+            deviceConfig.setDeviceType("Cisco-IOS");
+            deviceConfig.setVersion(i);
+            deviceConfigDao.saveOrUpdate(deviceConfig);
+        }
+    }
+
+    private void populateIpInterfaceAndGet() {
```

Comment:
The method does not return something. Maybe it should be renamed into `populateIpInterface`.